tests/realtikvtest/sessiontest/session_test.go

```diff
@@ -1021,3 +1041,928 @@ func TestDatabase(t *testing.T) {
+	tk = testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("set global tidb_enable_local_txn = on;")
+	tk.MustQuery("select @@txn_scope;").Check(testkit.Rows(kv.GlobalTxnScope))
+	require.Equal(t, kv.GlobalTxnScope, tk.Session().GetSessionVars().CheckAndGetTxnScope())
+	// Set @@txn_scope to local.
+	err = tk.ExecToErr("set @@txn_scope = 'local';")
+	require.Error(t, err)
+	require.Regexp(t, `.*txn_scope can not be set to local when zone label is empty or "global".*`, err)
+	tk.MustQuery("select @@txn_scope;").Check(testkit.Rows(kv.GlobalTxnScope))
+	require.Equal(t, kv.GlobalTxnScope, tk.Session().GetSessionVars().CheckAndGetTxnScope())
+	// Set @@txn_scope to global.
+	tk.MustExec("set @@txn_scope = 'global';")
+	tk.MustQuery("select @@txn_scope;").Check(testkit.Rows(kv.GlobalTxnScope))
+	require.Equal(t, kv.GlobalTxnScope, tk.Session().GetSessionVars().CheckAndGetTxnScope())
+
+	// @@tidb_enable_local_txn is on with configuring the zone label.
+	require.NoError(t, failpoint.Enable("tikvclient/injectTxnScope", `return("bj")`))
+	tk = testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("set global tidb_enable_local_txn = on;")
+	tk.MustQuery("select @@txn_scope;").Check(testkit.Rows(kv.LocalTxnScope))
+	require.Equal(t, "bj", tk.Session().GetSessionVars().CheckAndGetTxnScope())
+	// Set @@txn_scope to global.
+	tk.MustExec("set @@txn_scope = 'global';")
+	tk.MustQuery("select @@txn_scope;").Check(testkit.Rows(kv.GlobalTxnScope))
+	require.Equal(t, kv.GlobalTxnScope, tk.Session().GetSessionVars().CheckAndGetTxnScope())
+	// Set @@txn_scope to local.
+	tk.MustExec("set @@txn_scope = 'local';")
+	tk.MustQuery("select @@txn_scope;").Check(testkit.Rows(kv.LocalTxnScope))
+	require.Equal(t, "bj", tk.Session().GetSessionVars().CheckAndGetTxnScope())
+	// Try to set @@txn_scope to an invalid value.
+	err = tk.ExecToErr("set @@txn_scope='foo'")
+	require.Error(t, err)
+	require.Regexp(t, `.*txn_scope value should be global or local.*`, err)
+	require.NoError(t, failpoint.Disable("tikvclient/injectTxnScope"))
+}
+
+func TestDoDDLJobQuit(t *testing.T) {
+	// test https://github.com/pingcap/tidb/issues/18714, imitate DM's use environment
+	// use isolated store, because in below failpoint we will cancel its context
+	store, err := mockstore.NewMockStore(mockstore.WithStoreType(mockstore.MockTiKV))
+	require.NoError(t, err)
+	defer store.Close()
+	dom, err := session.BootstrapSession(store)
+	require.NoError(t, err)
+	defer dom.Close()
+	se, err := session.CreateSession(store)
+	require.NoError(t, err)
+	defer se.Close()
+
+	require.Nil(t, failpoint.Enable("github.com/pingcap/tidb/ddl/storeCloseInLoop", `return`))
+	defer failpoint.Disable("github.com/pingcap/tidb/ddl/storeCloseInLoop")
+
+	// this DDL call will enter deadloop before this fix
+	err = dom.DDL().CreateSchema(se, model.NewCIStr("testschema"), nil, nil)
+	require.Equal(t, "context canceled", err.Error())
+}
+
+func TestTemporaryTableInterceptor(t *testing.T) {
```

Comment:
This test doesn't belong to `testSessionSerialSuite` , would be covered by #34664, and you don't remove its counter part in session_legacy_test.go.

Please review all your changes to keep modification limited to `testSessionSerialSuite`.

Reply:
Done